### PR TITLE
[Proto] Qadence Tree

### LINKED ISSUE #399: [Proto] Qadence Tree
A prototype to replace whats now the `Converted` class to store all relationships between abstract,native and parameters with a dag which can be mapped over.  Whats now under 'qadence/tree.py' could for example be added to the QuantumCircuit class. lmkwyt

### PR DESCRIPTION
**Pull request by dominikandreasseitz**
Closes #399

### CODE REVIEW

File: qadence/tree.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+def converter_factory(backend_name: str) -> Callable:
+    module_path = f"qadence.backends.{backend_name}.convert_ops"
+    convert_block_fn: Callable
+    try:
+        module = importlib.import_module(module_path)
+        convert_block_fn = getattr(module, "convert_block")
+
+    except (ImportError, ModuleNotFoundError):
+        pass
```
Comment (Roland-djee) [inline qadence/tree.py:24]: Do we really pass here ?

**Review by Roland-djee — COMMENTED** [review]
Thanks @dominikandreasseitz looks fine as a basis. We should discuss this in the new year on how to expand on this and what kind of functionality we need to include.

**Comment by jpmoutinho** [comment]
Closing given the issue created in the new expression system.

**Comment by dominikandreasseitz** [comment]
archiving this @Roland-djee

**Pull request closed without merging**